### DISCUSSION TITLE
BirdNET-Pi HAT

### DISCUSSION THREAD

**Discussion by CaiusX** [mcguirepr89/BirdNET-Pi#77] [Ideas]
Hi All

I had an idea on Sunday morning to use a BLOKUSLabs [PiSound HAT](https://blokas.io/pisound/) which I had gathering dust as a sound capture board for BirdNET-Pi system.  The PiSound [kernel](https://github.com/BlokasLabs/pisound) hadn't been built for the 64bit system, so having enquired with the wizard @mcguirepr89, he had built the necessary libraries before I'd finished lunch on Sunday (notwithstanding him being 8 hours behind Africa timezone wise).  It took me an extra day to solder jumpers onto the PiSound and find a Pi who's GPIO's worked properly.  Having connected it all up Patrick got it working in BirdNET-Pi instantly.

The results are I think very good - the temporary installation using a "Pluggy" EM272 below:
![BirdNet-PiSound-Front - Copy](https://user-images.githubusercontent.com/25373276/143014400-3f2dbcc8-5baf-4742-ab0a-5bc252f5f33a.jpg)
![BirdNet-PiSound-Rear - Copy](https://user-images.githubusercontent.com/25373276/143014438-59d13e13-655d-4fb4-a501-241872e57636.jpg)

Gives the sounds:

https://user-images.githubusercontent.com/25373276/143014544-e581d2e0-b743-4345-b5bf-31d882b699ff.mp4

which BirdNET-Pi recognizes as:
![image](https://user-images.githubusercontent.com/25373276/143014608-53f7ffc0-8c01-42e4-ac1f-cb46441d4596.png)

![image](https://user-images.githubusercontent.com/25373276/143014752-14930e99-2a1f-4e03-80f2-0ec017a23473.png)

Whilst the PiSound-Mic combination is a bit of overkill, with Burr-Brown op-amp, DAC and midi functionality, I think there is potential to build a Pi-HAT, utilizing @DD4WH excellent work [HERE](https://github.com/DD4WH/SASS/wiki/Stereo-recording-low-noise-low-cost-system) and discussions  #39. 

Since the HAT will have AD conversion other measurements can also be gathered (temperature, humidity, pressure, air quality, rainfall, location, camera.....) and logged together with the BirdNET-Pi data; making this potentially a portable integrated nature measurement station.

I hope to prototype a variation of @DD4WH s

![image](https://user-images.githubusercontent.com/25373276/143019122-0789f8c3-fd92-43b8-9e73-7bf9e07d6331.png)

which can operate on the Pi's +5V on another board I have lying around

![PiHat - Copy](https://user-images.githubusercontent.com/25373276/143016613-35d8ed12-7020-4213-a747-882f0e1df2a1.jpg)

I will report progress as I go and will need a lot of help and advice on both hardware and software.

Best
AP

**Comment by mcguirepr89** [comment]
For what it is worth, I have put together some ESP-8266 weather gadgets before, and the DHT11 for temperature and humidity is pretty worthless for getting accurate readings. (It's great if you just need a trigger for drastic changes). My budget prevented me from testing anything with a better reputation, so I can't really recommend anything in particular - I just didn't want you wasting any time with a DHT11.

**Comment by DD4WH** [comment]
@CaiusX 
AP, that looks like a very good setup for high quality (stereo?) recordings of birds. So I would not say it is overkill. :-)
Unfortunately, I could not find a schematic or an indication which chips are used for the ADC in that PiSound-HAT. But the specs table says it has 40dB of (analog?) gain. So I think this setup could be as good as a DIY ultra low noise preamp plus EM272/EOM-5024 and a standard USB audio interface. However, to get the most out of it, I would recommend to use a 6.3mm extension/adapter to 3.5mm cable and put the Pluggy EM272 mic away from any interference of the RPi. But this is purely theoretical. Maybe you did not experience any digital or hum noise captured by the mic. If the latter is the case, everything is fine.

The plan to use the empty DIY RPi HAT-PCB board for the preamp is fine. Just a few words of caution:

* the HAT board will be plugged into the RPi and is therefore 1. very close to all the digital noise from the RPi (which could induce EMI into the preamp circuit) and 2. has several common ground connections with the RPi (with the possibility of causing ground loops) --> maybe think about using an extension cable and a separate shielded box for the preamp to suppress induced noise from the RPi

* the Proper preamp (ESP Elliot systems) is designed for 9V. It will probably be feasible to modify it for 5 V. But if you use the 5V from the RPi switching power supply adapter, you will have to spend considerable effort to filter the power supply. It could well be that it will not at all be possible to get rid of all the switching noise of the power supply. The preamp is ultra low noise and needs an ultra low noise power supply. So, my recommendation would be to use a separate power supply with a transformer-based AC supply, not a switching power supply. But I may be wrong and you can manage to filter out the noise, I have no concrete experimental evidence, just some experience with very dirty switching power supplies ;-).

All the best!

**Reply by CaiusX** [reply]
Thanks for the advice and insight @DD4WH 

My transistors have eventually landed in ZA & I am now able to try prototype the ESP system - could I get a picture of the back of your perfboard to make sure my wiring is right?

Any ideas what would need to change in the resistors/capacitors when reducing the voltage to 5V?

Best!

**Reply by DD4WH** [reply]
I have added a picture of the back of the perfboard into the WIKI, it is difficult to see the gaps in the board, check everything twice. My perfboard is a bit messy, I double-checked every possible connection before applying power for short circuits by using a Digital Ohmmeter. Note that it is a stereo circuit, so I use two circuits, one for each channel.
https://github.com/DD4WH/SASS/wiki/Stereo-recording-low-noise-low-cost-system
Not sure what exactly has to change when powered by 5 V. This is the original circuit:
https://sound-au.com/project93.htm

Be sure to:
* omit D1
* omit R10 & R11
* change polarity in C5 & C6
* change R2 to something like 1K or 470 Ohms --> electret capsule should have about 5 - 9 Volts
* I also omitted C6, R13 and F2 and used connection 1 (GND) and connection 2 as the audio output which is fed into the ADC of the USB audio interface

Good luck with soldering and building!
